stingray/stats.py

```diff
@@ -776,3 +786,332 @@ def _pavnosigfun(power, nspec):
         m -= 1
 
     return sum
+
+
+def power_confidence_limits(preal, n=1, alpha=0.16):
+    """Confidence limits on power, given a signal power in the PDS/Z search.
+
+    Adapted from Vaughan et al. 1994, noting that, after appropriate
+    normalization of the spectral stats, the distribution of powers in the PDS
+    and the Z^2_n searches is always described by a noncentral chi squared
+    distribution.
+
+    Parameters
+    ----------
+    preal: float
+        The underlying, signal-generated value of power
+
+    Other Parameters
+    ----------------
+    n: int
+        The number of summed powers to obtain pmeas. It can be multiple
+        harmonics of the PDS, adjacent bins in a PDS summed to collect all the
+        power in a QPO, or the n in Z^2_n
+    alpha: float
+        The p-value (e.g. 0.16 = 16% for 1-sigma)
```

Comment:
I am confused. If you're calculating confidence intervals, how do p-values factor in here?